echopype/convert/set_groups_ek80.py

```diff
@@ -1311,31 +1318,88 @@ def set_vendor(self) -> xr.Dataset:
+                coeffs_and_decimation[PULSE_COMPRESS][DECIMATION].append(
+                    self.parser_obj.fil_df[ch][2]
+                )
 
         # Assemble everything into a Dataset
         ds = xr.merge([ds_table, ds_cal])
 
-        # Save filter coefficients as real and imaginary parts as attributes
-        for k, v in coeffs.items():
-            ds.attrs[k + "_r"] = np.real(v)
-            ds.attrs[k + "_i"] = np.imag(v)
-
-        # Save decimation factors as attributes
-        for k, v in decimation_factors.items():
-            ds.attrs[k] = v
+        # Add the coeffs and decimation
+        ds = ds.pipe(self._add_cd, coeffs_and_decimation)
 
         # Save the entire config XML in vendor group in case of info loss
         ds.attrs["config_xml"] = self.parser_obj.config_datagram["xml"]
 
         return ds
+
+    @staticmethod
+    def _add_cd(
+        dataset: xr.Dataset, coeffs_and_decimation: Dict[str, Dict[str, List[Union[int, NDArray]]]]
+    ) -> xr.Dataset:
+        """
+        Assembles filter coefficient and decimation factors and add to the dataset
+
+        Parameters
+        ----------
+        dataset : xr.Dataset
+            xarray dataset where the filter coefficient and decimation factors will be added
+        coeffs_and_decimation : dict
+            dictionary holding the filter coefficient and decimation factors
+
+        Returns
+        -------
+        xr.Dataset
+            The modified dataset with filter coefficient and decimation factors included
+        """
+        attribute_values = {
+            FILTER_IMAG: "filter coefficients (imaginary part)",
+            FILTER_REAL: "filter coefficients (real part)",
+            DECIMATION: "decimation factor",
+            WIDE_BAND_TRANS: "Wideband transceiver",
+            PULSE_COMPRESS: "Pulse compression",
+        }
+
+        coeffs_xr_data = {}
+        for cd_type, values in coeffs_and_decimation.items():
+            for key, data in values.items():
+                if "filter" in key:
+                    attrs = {"long_name": f"{attribute_values[cd_type]} {attribute_values[key]}"}
+                    # filter_i and filter_r
+                    max_len = np.max([len(a) for a in data])
+                    # Pad arrays
+                    data = np.asarray(
+                        [
+                            np.pad(a, (0, max_len - len(a)), "constant", constant_values=np.nan)
+                            for a in data
+                        ]
+                    )
+                    dims = ["channel", f"{cd_type}_filter_n"]
+                else:
+                    attrs = {
+                        "long_name": f"{attribute_values[cd_type]} {attribute_values[DECIMATION]}"
+                    }
+                    dims = ["channel"]
+                # Set the xarray data dictionary
+                coeffs_xr_data[f"{cd_type} {key}"] = (dims, data, attrs)
```

Comment:
From the print out you showed, there's a space instead of an underscore between PC or WBT and filter_*. Seems better to replace it?
```suggestion
                coeffs_xr_data[f"{cd_type}_{key}"] = (dims, data, attrs)
```